buildSrc/build.gradle.kts

```diff
@@ -88,6 +89,18 @@ dependencies {
     compile("com.github.javaparser:javaparser-core:2.4.0")
 }
 
+// Allow Kotlin types to reference both Java and Groovy types
+// Remove compileGroovy -> compileJava dependency added by GroovyBasePlugin
+// Prevent cycles in the task graph as compileJava depends on compileKotlin
```

Comment:
This should really be fixed on the Kotlin side. Compiling Java should not depend on compiling Kotlin. Also, compiling Groovy should not depend on compiling Java by default or there should at least be some higher-level model that task.dependsOn for this.

`dependsOn.remove` is something that is exposed to the DSL by accident and can't be easily forbidden without deprecation, but it's not a good way of modeling this language relationship.

Reply:
> Compiling Java should not depend on compiling Kotlin.

I don't see any way of getting rid of this relationship. In order for the Java compiler to be aware of the Kotlin code it needs to be in bytecode format first.

Unless you're proposing keeping the dependency but modeling it with a different paradigm?